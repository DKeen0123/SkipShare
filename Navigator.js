import React, { Component } from 'react';
import { Text, View } from 'react-native';
import { createStackNavigator } from 'react-navigation';

class Home extends Component {
  render() {
    return (
      <View>
        <Text> textInComponent </Text>
      </View>
    )
  }
}

const Navigator = createStackNavigator({
  Home: {
    screen: Home
  }
});

const NavWrapper = () => {
  return <Navigator />;
}

export default NavWrapper;