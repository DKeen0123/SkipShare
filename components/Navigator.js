import React, { Component } from 'react';
import { Text, View, ActivityIndicator } from 'react-native';
import { createStackNavigator } from 'react-navigation';
import { graphql } from 'react-apollo';
import gql from 'graphql-tag';

import Login from './user/Login';

class Home extends Component {
  render() {
    return (
      <View>
        <Text> textInComponent </Text>
      </View>
    )
  }
}

const Navigator = createStackNavigator({
  Home: {
    screen: Home
  }
});

const NavWrapper = ({ loading,user }) => {
  console.log(user)
  if (loading) return <ActivityIndicator size="large" />;
  if (!user) return <Login />;
  return <Navigator />;
}

const userQuery = gql`
  query userQuery {
    user {
      id
      email
    }
  }
`
export default graphql(userQuery, {
  props: ({ data }) => ({ ...data })
})(NavWrapper);