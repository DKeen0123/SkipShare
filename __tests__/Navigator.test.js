import React from 'react';
import { shallow } from 'enzyme';
import Navigator from '../components/Navigator';

const navigator = shallow(<Navigator />);

it('renders without crashing', () => {
  expect(navigator).toMatchSnapshot();
});