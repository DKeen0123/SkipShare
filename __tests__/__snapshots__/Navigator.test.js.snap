// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Query
  displayName="Apollo(NavWrapper)"
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "userQuery",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "user",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "email",
                      },
                      "selectionSet": undefined,
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 63,
        "start": 0,
      },
    }
  }
  skip={false}
  warnUnhandledError={true}
/>
`;
